### PR TITLE
Improve packing of control flow graph.

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@cston, @mavasani, @dotnet/roslyn-compiler, @dotnet/analyzer-ioperation Please review.

File: src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs [hunk 1]
```diff
@@ -827,13 +845,19 @@ private static bool PackBlocks(ArrayBuilder<BasicBlock> blocks, PooledDictionary
                         count--;
                         removeBlock(block, currentRegion);
                         anyRemoved = true;
```
[the remark this answers is not in the document]
Comment (AlekseyTs) [reply src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs]: > Are anyRemoved and retry the same variable? I can't find any condition in which anyRemoved would be different from retry.

```retry``` is getting reset to false, but ```anyRemoved``` is not getting reset.

**Comment by AlekseyTs** [comment]
@cston, @dotnet/roslyn-compiler Please review.

**Comment by AlekseyTs** [comment]
@cston, @dotnet/roslyn-compiler Please review.

File: src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs [hunk 2]
```diff
@@ -537,13 +537,29 @@ private static bool PackBlocks(ArrayBuilder<BasicBlock> blocks, PooledDictionary
             ArrayBuilder<RegionBuilder> fromPredecessor = null;
 
             bool anyRemoved = false;
+retryLabel:
+            bool retry = false;
```
[the remark this answers is not in the document]
Comment (AlekseyTs) [reply src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs]: > Consider using a loop.

Sure.

---
In reply to: [183085566](https://github.com/dotnet/roslyn/pull/26234#discussion_r183085566) [](ancestors = 183085566)

**Comment by jcouv** [comment]
@AlekseyTs It looks like this is ready to merge.

**Comment by jcouv** [comment]
Oh, never mind. I see a pending change (converting goto's to loop).

**Pull request merged**